src/orders/components/OrderCancelDialog/OrderCancelDialog.test.tsx

```diff
@@ -0,0 +1,76 @@
+import { OrderErrorCode, OrderErrorFragment } from "@dashboard/graphql";
+import { render, screen } from "@testing-library/react";
+import React from "react";
+import { FormattedMessageProps } from "react-intl";
+
+import { OrderCancelDialog } from "./OrderCancelDialog";
+
+jest.mock("react-intl", () => ({
+  useIntl: jest.fn(() => ({
+    formatMessage: jest.fn(x => x.defaultMessage),
+  })),
+  defineMessages: jest.fn(x => x),
+  FormattedMessage: ({ defaultMessage, values }: FormattedMessageProps) => {
+    let formattedMessage = defaultMessage;
+
+    // Replace placeholders with corresponding values if values are provided
+    if (values) {
+      Object.keys(values).forEach(key => {
+        const valueRegex = new RegExp(`#{${key}}`, "g");
+        formattedMessage = formattedMessage.replace(
+          valueRegex,
+          values[key] as string,
+        );
+      });
+    }
```

Comment:
**issue**: this is not a mock, this is a new logic that actually requires a test. When you mock you manipulate something entirely, and manipulate the results eg. returns